src/plugins/ui_stateful/sheetview.ts

```diff
@@ -817,35 +831,91 @@ export class SheetViewPlugin extends UIPlugin {
     }
   }
 
-  getVisibleFigures(): Figure[] {
+  /**
+   * Return the index of a col given a negative offset x, based on the main viewport top
+   * visible cell of the main viewport.
+   * It returns -1 if no col is found.
+   */
+  private getColIndexLeftOfMainViewport(x: Pixel): HeaderIndex {
+    if (x >= 0) return -1;
     const sheetId = this.getters.getActiveSheetId();
-    const result: Figure[] = [];
+    let col = this.getActiveMainViewport().left;
+    let colStart =
+      -this.getActiveSheetScrollInfo().scrollX - this.getters.getColRowOffset("COL", col, 0);
+    for (; x < colStart; col--) {
+      colStart -= this.getters.getColSize(sheetId, col - 1);
+    }
+    return Math.max(col, 0);
+  }
+
+  /**
+   * Return the index of a row given a negative offset y, based on the main viewport top
+   * visible cell of the main viewport.
+   * It returns -1 if no row is found.
+   */
+  private getRowIndexTopOfMainViewport(y: Pixel): HeaderIndex {
+    if (y >= 0) return -1;
+    const sheetId = this.getters.getActiveSheetId();
+    let row = this.getActiveMainViewport().top;
+    let rowStart =
+      -this.getActiveSheetScrollInfo().scrollY - this.getters.getColRowOffset("ROW", row, 0);
+    for (; y < rowStart; row--) {
+      rowStart -= this.getters.getColSize(sheetId, row - 1);
```

Comment:
```suggestion
      rowStart -= this.getters.getRowSize(sheetId, row - 1);
```
or probably getUserRowSize ?